gradle-sls-packaging/src/main/groovy/com/palantir/gradle/dist/MaximumVersionComparator.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * (c) Copyright 2018 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.dist;
+
+import com.palantir.sls.versions.SlsVersionMatcher;
+import java.util.Comparator;
+import java.util.OptionalInt;
+
+enum MaximumVersionComparator implements Comparator<MaximumVersion> {
+    INSTANCE;
+
+    static final Comparator<OptionalInt> EMPTY_IS_GREATER =
+            Comparator.comparingInt(num -> num.isPresent() ? num.getAsInt() : Integer.MAX_VALUE);
+
+    static final Comparator<SlsVersionMatcher> MATCHER_COMPARATOR = Comparator
```

Comment:
This seems like a really nice pure function for dealing with version ranges.  Can we just put it in sls-versions now that it's open source?

Reply:
https://github.com/palantir/sls-version-java/pull/3